hugo/content/2-models/introduction.md

```diff
@@ -0,0 +1,166 @@
+
+## Specifying Fields to Serialize
+
+Before you can use a model, you must specify which of its fields you want to be included when an instance of a model is serialized and saved to the database. You can do this with the `MANGROVE_MAKE_KEYS_MODEL` macro.
+
+This macro accepts the class you are registering as a model, as well as a variadic list of the fields you want to include, wrapped as name-value pairs (**NVPs**).
+
+```cpp
+// Example of registering fields in model
+class Message : mangrove::model<Message> {
+    bsoncxx::oid author_id;
+    std::string content;
+    std::chrono::system_clock::time_point time_sent;
+    int dont_include_me;
+
+    MANGROVE_MAKE_KEYS_MODEL(Message,
+                             MANGROVE_NVP(author_id),
+                             MANGROVE_NVP(content),
+                             MANGROVE_NVP(time_sent))
+}
+```
+
+The `MANGROVE_NVP` macro specifies that you want its argument serialized in such a way that its **name** or **key** is the name of the variable. For example, in the above code sample, the resulting saved BSON document would look something like the following:
+
+```json
+{
+    "_id" : ObjectId("576d7238e249391db3271e31"),
+    "author_id" : ObjectId("576d5145e2493917691144f1"),
+    "content" : "Hey, did you see how cool Mangrove is?!",
+    "time_sent" : ISODate("2016-07-01T14:42:06.018Z")
+}
+```
+
+{{% notice info %}}
+While there is no limitation to the complexity of the classes you can turn into a model, there are limitations regarding which types are serializable and can be included in the `MANGROVE_MAKE_KEYS_MODEL` macro. These limitations are discussed in the [next section](/2-models/allowed-types). 
+{{% /notice %}}
+
+If you don't want the name of the field to be exactly the same as the name of your variable, you can use the macro `MANGROVE_CUSTOM_NVP` (TODO: may be different) to specify a custom name for the resulting BSON element of a particular field.
+
+```cpp
+// Example of registering fields in model with custom names
+class Message : mangrove::model<Message> {
+    bsoncxx::oid author_id;
+    std::string content;
+    std::chrono::system_clock::time_point time_sent;
+    int dont_include_me;
+
+    MANGROVE_MAKE_KEYS_MODEL(Message,
+                             MANGROVE_CUSTOM_NVP("a_id", author_id),
+                             MANGROVE_CUSTOM_NVP("c", content),
+                             MANGROVE_NVP(time_sent))
+}
+```
+
+In the above example, the resulting BSON documents would now look like the following:
+
+```json
+{
+    "_id" : ObjectId("576d7238e249391db3271e31"),
+    "a_id" : ObjectId("576d5145e2493917691144f1"),
+    "c" : "Hey, did you see how cool Mangrove is?!",
+    "time_sent" : ISODate("2016-07-01T14:42:06.018Z")
+}
+```
+
+{{% notice tip %}}
+
+If you have extremely large collections and want to save some disk space and bandwidth, you can use custom NVPs to shorten the names of fields associated with large variable names to a single-letter.
```

Comment:
That's a pretty interesting use case. ODM's do make it really painless to obfuscate / shorten your key names :)